### PR TITLE
feat(Import.CSV): Ignore lines from the start and/or end of csv file

### PR DESCRIPTION
**Pull request by gitys**
- Add row range
  - Add "ignore from start" and "ignore from end" num picker.
  - Rows not in import range will be grayed out in list.
  - Add row to ignore to template.
- Change "search" button to "browse"
- Remove limit of 10 line on import in preview table.
- Remove bold font
- Set max rows in preview to 20.
- Modified resizer proportions to prefer enlarging the list.
- Make log file read only.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by vomikan** [comment]
This branch has conflicts that must be resolved!
Nice Changes. Could you please resolve conflicts?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by gitys** [comment]
Resolved conflicts + shortened title to be <=72 chars

File: src/import_export/univcsvdialog.cpp [hunk 1]
```diff
@@ -71,6 +71,8 @@ mmUnivCSVDialog::mmUnivCSVDialog(
     is_importer_(is_importer),
     delimit_(","),
     importSuccessful_(false),
+	m_spinIgnoreFirstRows_(nullptr),
```
Comment (vomikan) [inline src/import_export/univcsvdialog.cpp]: please change the editor settings. Use breaks not tabs.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

 * Commit: 913662f5f9134655f79fa3d48fa57aa35ea90e54
  * Your commit message body contains a line that is longer than 72 characters
  * Your subject line is longer than 72 characters

 * Commit: 24077e4378520330f49cf59cb4fd87e16baa042c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: f73a811006810aecfc9747144b82e630e270cf3e
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, cleanup, format**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

 * Commit: 913662f5f9134655f79fa3d48fa57aa35ea90e54
  * Your commit message body contains a line that is longer than 72 characters
  * Your subject line is longer than 72 characters

 * Commit: 24077e4378520330f49cf59cb4fd87e16baa042c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: f73a811006810aecfc9747144b82e630e270cf3e
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, cleanup, format**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by gitys** [comment]
I see git cop is very upset :-(
- What is the convention for a merge commit?
- Look like spaces are illegal in scope. Maybe add that to wiki.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

 * Commit: 913662f5f9134655f79fa3d48fa57aa35ea90e54
  * Your commit message body contains a line that is longer than 72 characters
  * Your subject line is longer than 72 characters

 * Commit: 24077e4378520330f49cf59cb4fd87e16baa042c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: f73a811006810aecfc9747144b82e630e270cf3e
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, cleanup, format**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 346086c1329defb479455cb7a9dbd07464c622ba
  * Your subject line is longer than 72 characters

 * Commit: 913662f5f9134655f79fa3d48fa57aa35ea90e54
  * Your commit message body contains a line that is longer than 72 characters
  * Your subject line is longer than 72 characters

 * Commit: 24077e4378520330f49cf59cb4fd87e16baa042c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: f73a811006810aecfc9747144b82e630e270cf3e
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, cleanup, format**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by gitys** [comment]
Is there a way to push new commits to a new PullRequest?

**Comment by stef145g** [comment]
**GitCop** examines the commit subject line only.

- All lines must be less than 72 characters.

**Format is as follows:**
type**(**Scope.Of.Patch**):** description

**type** - Following pre-defined words.
- feat, fix, docs, style, refactor, perf, test, chore, revert, cleanup, format

**(Scope.Of.Patch)** - cannot contain spaces. Uppercase characters allowed.

**description** - this is the title for the patch using lower case only
- no punctuation characters allowed in the description title
- a colon and space must be added to separate the description field
- Extra lines can be added as necessary, with no limitation except for line length.

----------------------
if this description is better than what is in the current WIKI, I can update the WIKI to this.

**Comment by stef145g** [comment]
@gitys, In order to fix a current pull request, the only way I know is to close this and create a new pull request. This way patch titles can be fixed according to GitCop.
I refer you to the section "Correcting Our Mistakes in a Git Repository" in this description:
http://sourceforge.net/p/moneymanagerex/wiki/How%20to%20Develop%20Using%20Git/

As for adding extra info, if you already have an active pull request, any new commits to that pull request are automatically added to the existing pull request.

When other users add patches to your pull request, correcting your work and maintaining author ownership can be a little more tricky, but is still possible . In this case it may be best to maintain the existing pull request.

**Comment by gitys** [comment]
@stef145g, I think your description above is more detailed than what appears in the wiki and think it should be added in.

Regarding this pull request, even though not directly stated, it is clear to me that it is not being accepted because of the git cop issues. To fix this, as you say, the process will be (titled the "Destructive Option" in the link you sent):
- Cancel this pull request.
- Store all my commits in diffs. 
- Reset my branch to match main branch
- Reapply all my commits and merges from diffs.
- Force push to remote (something not very recommended to do in git)
- Create a new pull request.
- Merge again with whatever changes are made in the meantime.

The above process is expected to be very time consuming. I completely understand the significance of conventions but considering the amount of work required, I would like to ask if it is possible to overlook the git cop issue this time. There are 4 commits that have problems:
346086c and 913662f: 72 char line len- git hub seems to cut them anyway
24077e4: Missing ":"- my bad
f73a811: Invalid type- this is clearly wrong since "fix" is a valid type. Maybe the space in scope is the issue.

Looking back in commit history it is quite easy for find other non compliant commits so I assume this is possible. 
I'd be grateful if you can spare me the many hours of work required to fix this. This is my first pull request, I'll make sure to be compliant in following pull requests.

**Comment by gitys** [comment]
Thanks, I appreciate it.

**Comment by stef145g** [comment]
@gitys , My apologies if I gave the wrong impression. @vomikan has a better understanding of the affected code, and could test it, which is the reason I did not commit your code. My responses were an attempt to answer your questions and were meant as explanation only.

The explained procedure is good to fix or consolidate patches in a localised branch where only one user is accessing that branch. It is something I discovered a while ago, when things went wrong in my repo, and this method allowed me to fix my issues.

The GitCop explanation has now been updated and hopefully will be easier to understand for future reference.

**Comment by gitys** [comment]
@stef145g , no problem at all. I hope you find my changes to be beneficial to the program.

**Pull request merged**